pkg/mpc-examples/contracts/04-pause-unpause-controller/PauseUnpauseControllerFactory.sol

```diff
@@ -0,0 +1,128 @@
+    struct MinimalPoolParams {
+        string name;
+        string symbol;
+        IERC20[] tokens;
+        uint256[] normalizedWeights;
+        // uint256[] assetManagers;
+        uint256 swapFeePercentage;
+        bool swapEnabledOnStart;
+        bool mustAllowlistLPs;
+        uint256 managementAumFeePercentage;
+        uint256 aumFeeId;
+    }
+
+    event ControllerCreated(address indexed controller, bytes32 poolId);
+    event Disabled();
+
+    constructor(address factory, IVault vault) {
+        managedPoolFactory = IManagedPoolFactory(factory);
+        balancerVault = vault;
+    }
+
+    /// === Getters === ///
+    function getLastCreatedPool() external view returns (address) {
+        return _lastCreatedPool;
+    }
+
+    /// === Setters === ///
+
+    function create(MinimalPoolParams memory minimalParams, address controllerOwner) external {
+        // checks
+        require(!isDisabled, "Controller factory is disabled");
+        require(!managedPoolFactory.isDisabled(), "managed Pool factory disabled");
+
+        bytes32 controllerSalt = bytes32(_nextControllerSalt);
+        _nextControllerSalt += 1;
+
+        bytes memory controllerCreationCode = abi.encodePacked(
+            type(PauseUnpauseController).creationCode,
+            abi.encode(balancerVault, controllerOwner) //constructor args
+        );
+
+        address expectedControllerAddress = Create2.computeAddress(controllerSalt, keccak256(controllerCreationCode));
+
+        // The asset managers have the ability to manage poolTokens. Since this controller
+        // is focused on a secure pause/unpausing, allowing to freely pass asset managers
+        // during pool creation could undermine the narrative that a paused pool does not have
+        // any "token movements". Passing the controller as asset manager (which does not have
+        // any logic for that purpose) confirms that narrative for this example.
+        address[] memory assetManagers = new address[](minimalParams.tokens.length);
+        for (uint256 i = 0; i < assetManagers.length; i++) {
+            assetManagers[i] = expectedControllerAddress;
+        }
+
+        IManagedPoolFactory.NewPoolParams memory fullParams;
+        fullParams.name = minimalParams.name;
+        fullParams.symbol = minimalParams.symbol;
+        fullParams.tokens = minimalParams.tokens;
+        fullParams.normalizedWeights = minimalParams.normalizedWeights;
+        fullParams.assetManagers = assetManagers;
+        fullParams.swapFeePercentage = minimalParams.swapFeePercentage;
+        fullParams.swapEnabledOnStart = minimalParams.swapEnabledOnStart;
+        fullParams.mustAllowlistLPs = minimalParams.mustAllowlistLPs;
+        fullParams.managementAumFeePercentage = minimalParams.managementAumFeePercentage;
+        fullParams.aumFeeId = minimalParams.aumFeeId;
+
+        _lastCreatedPool = managedPoolFactory.create(fullParams, expectedControllerAddress);
+
```

Comment:
Im fine with how this is done, but the nullController sets it up a little bit differently. 
```
IManagedPool pool = IManagedPool(IManagedPoolFactory(managedPoolFactory).create(fullParams, expectedControllerAddress);
_lastCreatedPool = address(pool);
```

If its done this way you will need to make the global managedPoolFactory variable of type address.

Id check with @gerrrg or @rabmarut on this one

Reply:
@gerrrg What is your opinion here?